Dnn.CommunityForums/class/ForumController.cs

```diff
@@ -204,58 +204,61 @@ internal Forum Forums_Get(int portalId, int moduleId, int forumId, bool useCache
 
             return forumId <= 0 ? null : GetForum(portalId, moduleId, forumId, !useCache);
         }
+        [Obsolete("Deprecated in Community Forums. Scheduled removal in 10.00.00. Use Forums_Save(int portalId, Forum fi, bool isNew, bool useGroupFeatures, bool useGroupSecurity)")]
         public int Forums_Save(int portalId, Forum fi, bool isNew, bool useGroup)
         {
-            var rc = new RoleController();
-            var db = new Data.Common();
+            return Forums_Save(portalId, fi,  isNew, useGroup, useGroup);
+        }
+        public int Forums_Save(int portalId, Forum fi, bool isNew, bool useGroupFeatures, bool useGroupSecurity)
+        {
             var permissionsId = -1;
-            if (useGroup && (string.IsNullOrEmpty(fi.ForumSettingsKey) || fi.PermissionsId == -1))
+            var fc = new ForumGroupController();
+            var fg = fc.Groups_Get(fi.ModuleId, fi.ForumGroupId);
+            if (useGroupSecurity)
             {
-                var fc = new ForumGroupController();
-                var fg = fc.Groups_Get(fi.ModuleId, fi.ForumGroupId);
                 if (fg != null)
                 {
-                    fi.ForumSettingsKey = fg.GroupSettingsKey;
-                    //fi.ModuleId = fg.ModuleId
                     fi.PermissionsId = fg.PermissionsId;
                 }
             }
-            else if (fi.PermissionsId <= 0 && useGroup == false)
+            else 
             {
-                var ri = rc.GetRoleByName(portalId, "Administrators");
-                if (ri != null)
-                {
-                    fi.PermissionsId = db.CreatePermSet(ri.RoleID.ToString());
-                    permissionsId = fi.PermissionsId;
-                    isNew = true;
-                }
-                if (fi.ForumID > 0 & !(string.IsNullOrEmpty(fi.ForumSettingsKey)))
+                if (fi.PermissionsId <= 0 || fi.InheritSecurity) /* if new forum or changing from inheriting security, create new permissions set */
                 {
-                    if (fi.ForumSettingsKey.Contains("G:"))
-                        fi.ForumSettingsKey = string.Empty;
+                    var rc = new RoleController();
+                    var ri = rc.GetRoleByName(portalId, "Administrators");
```

Comment:
@johnhenley We should use the built-in value for this:  `PortalSettings.AdministratorRoleName`

Reply:
I agree but can we make that a separate issue? There are a number of these, and it's PortalSettings.Current.AdministratorRoleName. Since this is in a controller, PortalSettings.Current may not be valid, rght? So need to do a more thoughtful job on it than just changing it on the fly?

Reply:
> I agree but can we make that a separate issue? There are a number of these, and it's PortalSettings.Current.AdministratorRoleName. Since this is in a controller, PortalSettings.Current may not be valid, rght? So need to do a more thoughtful job on it than just changing it on the fly?

@WillStrohl I created a separate issue #684 for the strings/constants